### PR TITLE
feat(timeseriescard): allow multiple alerts to match the same timeseries datapoint

### PR DESCRIPTION
**Pull request by scottdickerson**
Closes #

**Summary**

Improve the tooltip and markup to allow multiple alert ranges to match a given datapoint for a TimeSeriesCard
image

**Change List (commits, features, bugs, etc)**

- fix(timeseriescard): fix style lint issue with the style
- feat(TimeSeriesCard): add support for multiple matching alert ranges for one datapoint
- feat(timeSeriesUtils): add support for matching multiple alert ranges

**Acceptance Test (how to verify the PR)**

- tests here

### CODE REVIEW

File: src/components/TimeSeriesCard/TimeSeriesCard.jsx [hunk 1]
```diff
@@ -263,7 +268,7 @@ const TimeSeriesCard = ({
     const defaultFillColor = !isEditable ? originalFillColor : '#f3f3f3';
     if (!isNil(data)) {
       const matchingAlertRange = findMatchingAlertRange(alertRanges, data);
-      return matchingAlertRange ? matchingAlertRange.color : defaultFillColor;
+      return matchingAlertRange?.length > 0 ? matchingAlertRange[0].color : defaultFillColor;
```
Comment (tay1orjones) [inline src/components/TimeSeriesCard/TimeSeriesCard.jsx]: Is our babel config set to enable optional chaining?

File: src/components/TimeSeriesCard/TimeSeriesCard.story.jsx [hunk 2]
```diff
@@ -993,7 +993,7 @@ storiesOf('Watson IoT|TimeSeriesCard', module)
                 details: 'Alert name',
               },
               {
-                startTimestamp: 1572804320000,
+                startTimestamp: 1572313622000,
```
Comment (tay1orjones) [inline src/components/TimeSeriesCard/TimeSeriesCard.story.jsx]: Despite this, I'm not seeing any alerts on the `highlight alert ranges` story?

image

File: src/components/TimeSeriesCard/TimeSeriesCard.jsx (hunk 1 above)
Comment (scottdickerson) [reply src/components/TimeSeriesCard/TimeSeriesCard.jsx]: it is!

File: src/components/TimeSeriesCard/TimeSeriesCard.story.jsx (hunk 2 above)
Comment (scottdickerson) [reply src/components/TimeSeriesCard/TimeSeriesCard.story.jsx]: there's a weird bug with the time series stories where sometimes the chart is rendering a previous chart.  Try and force refresh your page and you should see the Alert Ranges I think.

File: src/components/TimeSeriesCard/timeSeriesUtils.js [hunk 3]
```diff
@@ -116,18 +117,22 @@ export const formatGraphTick = (
     ? currentTimestamp.format(dailyFormat)
```
Comment (sstone2423) [inline src/components/TimeSeriesCard/timeSeriesUtils.js]: This is a ton of conditionals, but I'm not seeing a better way at the moment. Is it possible to only run some of  these checks if the language is Chinese?

Comment (scottdickerson) [reply src/components/TimeSeriesCard/timeSeriesUtils.js]: I could switch statement and work within there, but honestly I'm just changing the existing code to support week.

File: src/components/TimeSeriesCard/TimeSeriesCard.story.jsx (hunk 2 above)
Comment (tay1orjones) [reply src/components/TimeSeriesCard/TimeSeriesCard.story.jsx]: @scottdickerson Got it! Could you enter an issue for that timeseries chart rendering bug?

**Pull request merged**